### PR TITLE
Generate resource.rc file dynamically. Enable Autotools cross-compilation.

### PR DESCRIPTION
**Pull request by UncombedCoconut**
WARNING: Broken under autotools. TBH this is just exposing a problem that already existed -- the same usage of `@VMAJOR@.@VMINOR@.@VREV@` will cause extras/macos/bundle.sh to fail (or, rather, emit a bad Info.plist file).
We will need to solve this before release.
I think we do need a numeric major.minor.rev[.build] rendition of the semver simply because operating systems have fields for that.
If we do that centrally, some Windows scripts I and @ProjectSynchro can benefit. (They've been using the 4th numeric field for a nightly build's YMD.)

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -186,7 +189,9 @@ if buildExec.disabled() == false
 
    if host_machine.system() == 'windows'
       windows = import('windows')
-      naev_source += windows.compile_resources('extras/resource.rc')
+      icon = configure_file(input: 'extras/logos/logo.ico', output: 'logo.ico', copy: true)
```
Comment (nloewen) [inline meson.build:192]: Does this really need to be copied to the build directory? Does a files() object not work?

**Comment by UncombedCoconut** [comment]
Hmm, probably.
The resource.rc file must contain an icon path which is resolvable from the
build directory (where windres will compile it).
Nothing stops us from making that a @SUBSTITUTION@. However, Autotools must
be equally able to fill it in.
If a file resource is what I need to make that path substitution in Meson,
and I can tell Autotools to just substitute an absolute path, this probably
gets a whole lot cleaner.

**Comment by UncombedCoconut** [comment]
Sorry, not understanding. How does Windres correctly find the icon file in
this setup?

**Comment by nloewen** [comment]
It looks like depend_files adds the provided files to the windres search path. So referencing "icon.ico" in resources.rc and adding the source file to the search path works. I've tested this, and can confirm it works as desired. Just change line 192 to `icon = files('extras/logos/logo.ico')`.

**Comment by UncombedCoconut** [comment]
Thanks for the feedback! Re-presenting for review (with change history cleaned up).

File: extras/resource.rc.in [hunk 2]
```diff
@@ -25,7 +25,7 @@ BEGIN
 			VALUE "LegalCopyright", "Copyright (C) 2006-2020"
 			VALUE "OriginalFilename", "naev.exe"
 			VALUE "ProductName", "Naev: Sea of Darkness"
-			VALUE "ProductVersion", "0.8.0"
+			VALUE "ProductVersion", "@VMAJOR@.@VMINOR@.@VREV@"
```
Comment (bobbens) [inline extras/resource.rc.in:28]: Since this is a string, wouldn't just @VERSION@ suffice?

**Comment by UncombedCoconut** [comment]
According to MS's docs, the "ProductVersion" value is only valid if it's a
dotted triple of numbers.
Now, it's quite possible the OS will find a way to live with a more general
string like "0.8.0-beta.3" anyway, but I have not seen any promises around
that.
Similar concern for macos, BTW, with the Info.plist CFBundleVersion field.

**Comment by bobbens** [comment]
OK, thanks!

**Pull request merged**